src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseTestService.java

```diff
@@ -1835,4 +1840,41 @@ void exportSolutionRepository_shouldReturnFileOrForbidden() throws Exception {
         exportSolutionRepository(exerciseRepo, HttpStatus.FORBIDDEN);
 
     }
+
+    // TEST
+    public void buildLogStatistics_unauthorized() throws Exception {
+        exercise = programmingExerciseRepository.save(ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course));
+        request.get("/api/programming-exercises/" + exercise.getId() + "/build-log-statistics", HttpStatus.FORBIDDEN, BuildLogStatisticsDTO.class);
+    }
+
+    // TEST
+    public void buildLogStatistics_noStatistics() throws Exception {
+        exercise = programmingExerciseRepository.save(ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course));
+        var statistics = request.get("/api/programming-exercises/" + exercise.getId() + "/build-log-statistics", HttpStatus.OK, BuildLogStatisticsDTO.class);
+        assertThat(statistics.getBuildCount()).isEqualTo(0);
+        assertThat(statistics.getAgentSetupDuration()).isEqualTo(null);
```

Comment:
```suggestion
        assertThat(statistics.getAgentSetupDuration()).isNull();
```
I think there is a special assertion to check for `null`.